### PR TITLE
Queryparams string value

### PR DESCRIPTION
**Pull request by atdcloud**
#### What does this PR solve?
Add functionality for query params as a string. 

#### Final checklist
- [x] Implementation is aligned to Spec.
- [x] Tests - Add proper tests to the added code.

### CODE REVIEW

File: __TESTS__/unit/url/url.test.ts [hunk 1]
```diff
@@ -101,6 +101,12 @@ describe('Tests for URL configuration', () => {
     expect(url).toEqual(`https://res.cloudinary.com/demo/image/upload/sample?_i=abcde&_z=1234&_t=false`);
   });
 
+  it('Should include query params', function () {
```
Comment (pawelphilipczyk-cloudinary) [inline __TESTS__/unit/url/url.test.ts:104]:
```suggestion
  it('Should include query params passed as a string', function () {
```

File: src/assets/CloudinaryFile.ts [hunk 2]
```diff
@@ -283,15 +283,34 @@ class CloudinaryFile {
...
-      // urlConfig.analytics is true by default, has to be explicitly set to false to overwrite
-      // Don't add analytics when publicId includes a '?' to not risk changing existing query params
-      if (this.urlConfig.analytics !== false && !(publicID.includes('?'))) {
-        queryParams.set("_a", getSDKAnalyticsSignature(trackedAnalytics));
+      const shouldAddAnalytics = this.urlConfig.analytics !== false && !(publicID.includes('?'));    
+
+      let queryParamsString = '';
+
+      if (typeof(this.urlConfig.queryParams) === 'object'){ 
+
```
Comment (pawelphilipczyk-cloudinary) [inline src/assets/CloudinaryFile.ts:292]:
```suggestion
```

File: src/assets/CloudinaryFile.ts [hunk 3]
```diff
@@ -283,15 +283,34 @@ class CloudinaryFile {
...
+          const queryParams = new URLSearchParams(this.urlConfig.queryParams as Record<string, string>);
+          
+          if (shouldAddAnalytics) {
+            queryParams.set("_a", getSDKAnalyticsSignature(trackedAnalytics));
+          }
+         
+          queryParamsString = queryParams.toString();
+
+        } catch(err) {
+          console.error("Error: URLSearchParams is not available");
```
Comment (pawelphilipczyk-cloudinary) [inline src/assets/CloudinaryFile.ts:303]:
```suggestion
          // TODO: Consult the error message with docs team
          console.error("URLSearchParams is not available so the queryParams object cannot be parsed, please try passing in a form of a string instead");
```

File: src/assets/CloudinaryFile.ts [hunk 4]
```diff
@@ -283,15 +283,34 @@ class CloudinaryFile {
...
+        }
+      } 
+      else{ 
+        queryParamsString = this.urlConfig.queryParams || '';
+        
+        if (shouldAddAnalytics) {
+          queryParamsString += `${(queryParamsString.length > 0 ? "&" :"")}_a=${getSDKAnalyticsSignature(trackedAnalytics)}`;
```
Comment (pawelphilipczyk-cloudinary) [inline src/assets/CloudinaryFile.ts:310]:
```suggestion
          queryParamsString += `${(queryParamsString.length > 0 ? '&' :'')}_a=${getSDKAnalyticsSignature(trackedAnalytics)}`;
```

**Review by pawelphilipczyk-cloudinary — APPROVED** [review]
Looks good, let's consult with the SDK guild on that. 

I've added few minor code suggestions.

**Review by lbendavid-cloudinary — APPROVED** [review]

**Pull request merged**